ledger-core/virtual/handlers/vstatereport.go

```diff
@@ -60,14 +60,86 @@ func (s *SMVStateReport) Init(ctx smachine.InitializationContext) smachine.State
 	}
 
 	ctx.SetDefaultMigration(s.migrationDefault)
-	return ctx.Jump(s.stepProcess)
+	return ctx.Jump(s.stepValidate)
 }
 
 func (s *SMVStateReport) migrationDefault(ctx smachine.MigrationContext) smachine.StateUpdate {
 	ctx.Log().Trace("stop processing VStateReport since pulse was changed")
 	return ctx.Stop()
 }
 
+func (s *SMVStateReport) stepValidate(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	switch report := s.Payload; {
+	case report.GetObject().IsEmpty():
+		return ctx.Error(throw.New("Object should not be empty"))
+	case report.GetAsOf().IsUnknown():
+		return ctx.Error(throw.New("AsOf can not be Unknown"))
+	case !report.GetDelegationSpec().IsZero():
+		return ctx.Error(throw.New("DelegationSpec should be empty"))
+	default:
+		switch report.GetStatus() {
+		case payload.Unknown:
+			return ctx.Error(throw.New("Unexepected Unknown status received"))
+		case payload.Missing:
```

Comment:
`Inactive` and `Missing` are the same :)

Reply:
`Missing` can have a lot more checks, it means NotFound, so I'd check a lot more fields are empty